libraries/node-core-library/src/LockFile.ts

```diff
@@ -85,6 +85,28 @@ export class LockFile {
     throw new Error(`File locking not implemented for platform: "${process.platform}"`);
   }
 
+  /**
+   * Attempts to create the lockfile.
+   * Will continue to loop at the given interval until the lock becomes available.
+   * Note that this function is subject to starvation issues.
+   */
+  public static acquire(resourceDir: string, resourceName: string, interval?: number): Promise<LockFile> {
+    interval = interval || 100;
+
+    const tryLock: (resolve: (value: LockFile) => void) => void = (resolve => {
```

Comment:
>(resolve: (value: LockFile) => void) => void = (resolve => { [](start = 19, length = 60)


This is unintelligible.  Try to make it something more straightforward like this:
```
retryLoop(int startTime, int maxWaitMs): Promise {
  x = tryAcquireLock();
  if (x) return Promise.resolve(x);
  if (now() > startTime+maxWaitMs) {
    return Promise.reject(...);
  }
  return this.sleepForMs(100).then((startTime, waitMaxMs) => {
    return retryloop();
  });
}
```

The `new Promise` convolutions should be rolled up inside `sleepForMs()` with a nice little bow on it